### PR TITLE
fix(vite-plugin-vue-i18n): use correct id in hmr

### LINKED ISSUE #153: Live reloading translations is not working
### Reporting a bug?

We just upgraded to vite version 3 and vite-plugin-vue-i18n to version 5, and reloading translations is not working. We detected this problem by starting a new project, but also found this problem with an old project.

I've tested it with vite version 2 and vite-plugin-vue-i18n version 3 before upgrading and everything worked as expected, after upgrading both packages the translations won't reload. In the vite console, there is also no message the translations are modified. After re-running the vite dev server, the new translations are working.

### Expected behavior

I expect in the console to see the message:
12:34:14 PM [vite] page reload lang/nl.json

And also that the page is reloading.

### Reproduction

- Upgrade an already working project or create a new project with vite version 3 & vite-plugin-vue-i18n version 5
- Run the vite dev server
- Modify any translation

### Issue Package

vite-plugin-vue-i18n

### System Info

```shell
System:
    OS: macOS 12.4
    CPU: (8) arm64 Apple M1
    Memory: 294.42 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 14.19.2 - ~/.nvm/versions/node/v14.19.2/bin/node
    npm: 6.14.17 - ~/.nvm/versions/node/v14.19.2/bin/npm
  Browsers:
    Chrome: 103.0.5060.134
    Firefox Developer Edition: 103.0
    Safari: 15.5
  npmPackages:
    @intlify/vite-plugin-vue-i18n: ^5.0.0 => 5.0.1 
    vite: ^3.0.0 => 3.0.3 
    vue: ^3.2.37 => 3.2.37 
    vue-i18n: ^9.1.10 => 9.1.10
```

### Screenshot

_No response_

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/bundle-tools/blob/main/.github/contributing.md).
- [X] Read the README
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/intlify/bundle-tools/discussions).

### PR DESCRIPTION
**Pull request by marekvospel**
fixes #153 

I haven't tested this, as I currently don't have access to my computer, nor github codespaces.

### CODE REVIEW

**Comment by marekvospel** [comment]
According to @ghostiam's comment in #153, this should fix #153.

**Comment by c-harding** [comment]
@ghostiam’s comment: https://github.com/intlify/bundle-tools/issues/153#issuecomment-1209653696

File: packages/vite-plugin-vue-i18n/src/index.ts [hunk 1]
```diff
@@ -210,7 +210,9 @@ function pluginI18n(
     async handleHotUpdate({ file, server }) {
       if (/\.(json5?|ya?ml)$/.test(file)) {
         const module = server.moduleGraph.getModuleById(
-          INTLIFY_BUNDLE_IMPORT_ID
+          getVirtualId(
+            INTLIFY_BUNDLE_IMPORT_ID
+          )
```
Comment (c-harding) [inline packages/vite-plugin-vue-i18n/src/index.ts:215]:
```suggestion
          asVirtualId(INTLIFY_BUNDLE_IMPORT_ID)
```

getVirtualId removes the leading `\0`, possibly returning null. In this case we want to add `\0` (see @ghostiam’s comment at https://github.com/intlify/bundle-tools/issues/153#issuecomment-1209653696, `server.moduleGraph.getModuleById` only works with a leading `\0`).

I have tested this locally, and live reloading works when using `asVirtualId`.

**Review by c-harding — CHANGES_REQUESTED** [review]

File: packages/vite-plugin-vue-i18n/src/index.ts [hunk 2]
```diff
@@ -210,7 +210,9 @@ function pluginI18n(
     async handleHotUpdate({ file, server }) {
       if (/\.(json5?|ya?ml)$/.test(file)) {
         const module = server.moduleGraph.getModuleById(
-          INTLIFY_BUNDLE_IMPORT_ID
+          getVirtualId(
```
Comment (kazupon) [inline packages/vite-plugin-vue-i18n/src/index.ts:213]: Oh... I forgot that virtual id has been used in the plugin 😅

**Review by kazupon — APPROVED** [review]
Thank you very much!
Welcome contribution!

LGTM!

**Comment by kazupon** [comment]
@marekvospel 
Could you fix the lint error, please?

**Comment by c-harding** [comment]
@kazupon this currently fails building. I would suggest you add a build stage to GitHub Actions (I’m happy to do so with a PR)

```
src/index.ts(213,11): error TS2345: Argument of type 'string | null' is not assignable to parameter of type 'string'.
  Type 'null' is not assignable to type 'string'.
```

File: packages/vite-plugin-vue-i18n/src/index.ts (hunk 1 above)
Comment (marekvospel) [reply packages/vite-plugin-vue-i18n/src/index.ts:215]: mb, I have remembered the name of incorrect function.

**Comment by marekvospel** [comment]
I've fixed the lint problem & changed to the correct method, so build should pass

**Comment by kazupon** [comment]
@marekvospel 
Thanks!

**Pull request merged**